File: test/fakers/checkout/installment_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/checkout/dtos/installment_dto.dart';

class InstallmentFaker {
  static final _faker = Faker();

  static InstallmentDto fakeDto({
    int? number,
    String? value,
    String? totalValue,
    String? text,
    bool? interestFree,
  }) {
    final num = number ?? _faker.randomGenerator.integer(12, min: 1);
    final val = value ?? 'R\$ ${_faker.randomGenerator.decimal()}';

    return InstallmentDto(
      number: num,
      value: val,
      totalValue: totalValue ?? 'R\$ ${_faker.randomGenerator.decimal() * num}',
      text: text ?? '${num}x de $val',
      interestFree: interestFree ?? _faker.randomGenerator.boolean(),
    );
  }
}

File: src/core/checkout/entities/fakers/installment-faker.ts
import { faker } from "@faker-js/faker"

import type { Installment } from "../installment"

export class InstallmentFaker {
  static fake(installment?: Partial<Installment>): Installment {
    const number = faker.number.int({ min: 1, max: 12 })
    const value = faker.number.float({ fractionDigits: 2, min: 10, max: 500 })

    return {
      interestFree: faker.datatype.boolean(),
      number,
      text: `${number}x de R$ ${value.toFixed(2)}`,
      totalValue: `R$ ${(value * number).toFixed(2)}`,
      value: `R$ ${value.toFixed(2)}`,
      ...installment,
    }
  }

  static fakeMany(count = 10, installment?: Partial<Installment>): Installment[] {
    return Array.from({ length: count }, () => InstallmentFaker.fake(installment))
  }
}
